SABT_MainUnit/learn_letter_mode.c

```diff
@@ -0,0 +1,261 @@
+
+
+/*
+ * @file learn_letter_mode.c
+ *
+ * @brief code for helper functions "learning letter modes" (MD 2,7,8,12)
+ * @author: Edward Cai
+ */ 
+
+// Standard libraries
+#include <stdbool.h>
+#include <stdio.h>
+
+// Custom libraries
+#include "audio.h"
+#include "io.h"
+#include "common.h"
+#include "debug.h"
+#include "script_common.h"
+#include "learn_letter_mode.h"
+
+// State variables
+static char next_state = STATE_NULL;
+static char user_response = NO_DOTS;
+static char submode = SUBMODE_NULL;
+static int index = 0;
+static glyph_t* curr_glyph = NULL;
+static glyph_t* user_glyph = NULL;
+static char cell = 0;
+static char cell_pattern = 0;
+static char cell_control = 0;
+static int incorrect_tries = 0;
+static bool scrolled = false;
+
+void learn_letter_reset(script_t* new_script, char* new_lang_fileset, char* new_mode_fileset) {
+	set_mode_globals(new_script, new_lang_fileset, new_mode_fileset);
+	next_state = STATE_MENU;
+	user_response = NO_DOTS;
+	submode = SUBMODE_NULL;
+	index = 0;
+	curr_glyph = NULL;
+	user_glyph = NULL;
+	cell = 0;
+	cell_pattern = 0;
+	cell_control = 0;
+	incorrect_tries = 0;
+	scrolled = false;
+	PRINTF("[MD2] Mode reset\n\r");
```

Comment:
If possible, change [MD2] labels to something generic or more mode number specific, depending on what you want.